addons/website_sale_stock/static/src/js/website_sale_stock.js

```diff
@@ -0,0 +1,6 @@
+$(document).ready(function () {
+    //show warning outofstock, when click on add to cart from shop ,trigger on change of quantity input
+    if(window.location.pathname === '/shop/cart' && $('.oe_cart .show_outofstock_warning').length) {
+        $('.oe_cart input.js_quantity').trigger('change');
+    }
+});
```

Comment:
ok , then we can remove `window.location.pathname === '/shop/cart'` and relay on `$('.oe_cart .show_outofstock_warning').length)` , when we got class `.show_outofstock_warning` we trigger input , is it ok ?